### PR TITLE
Update explainer.md 

### PR DESCRIPTION
**Pull request by yutakahirano**
Reflect the interface changes #134 and #135.

Also delete trailing whitespaces.

### CODE REVIEW

File: explainer.md [hunk 1]
```diff
@@ -58,22 +58,22 @@ encrypted and congestion-controlled communication.
...
 ## Example of sending unreliable game state to server using QUIC datagrams
 
 ```javascript
 // The app provides a way to get a serialized state to send to the server
 function getSerializedGameState() { ... }
 
-const transport = new QuicTransport('example.com', 10001);
-const datagramWriter = transport.sendDatagrams().getWriter();
+const transport = new WebTransport('quic-transport://example.com:10001/path');
+const datagramWriter = transport.writableDatagrams.getWriter();
```
Comment (ricea) [inline explainer.md:76]: `datagramWritable` is what we've implemented in Blink.

File: explainer.md [hunk 2]
```diff
@@ -147,13 +152,13 @@ for await (const stream of transport.receiveBidirectionalStreams()) {
 
 ```javascript
 const mediaSource = new MediaSource();
-await new Promise(resolve => mediaSource.addEventListener('sourceopen', () => resolve(), {once: true}));
+await new Promise(resolve => mediaSource.addEventListener('sourceopen', resolve, {once: true}));
 const sourceBuffer = mediaSource.addSourceBuffer('video/webm; codecs="opus, vp09.00.10.08"');
-const transport = new Http3Transport("/video");
-await fetch('http://example.com/babyshark');
-for await (const datagram of transport.receiveDatagrams()) {
+const transport = new WebTransport('/video');
```
Comment (ricea) [inline explainer.md:157]: Interesting. I suppose we can make relative URLs work without too much difficulty. I wonder how useful it is, though.

File: explainer.md [hunk 3]
```diff
@@ -147,13 +152,13 @@ for await (const stream of transport.receiveBidirectionalStreams()) {
...
+await fetch('https://example.com/babyshark');
```
Comment (ricea) [inline explainer.md:158]: This is odd. Do you know what this fetch is doing here?

File: explainer.md [hunk 4]
```diff
@@ -147,13 +152,13 @@ for await (const stream of transport.receiveBidirectionalStreams()) {
...
+for await (const datagram of transport.readableDatagrams) {
```
Comment (ricea) [inline explainer.md:159]: `datagramReadable`?

File: explainer.md [hunk 5]
```diff
@@ -163,41 +168,46 @@ for await (const datagram of transport.receiveDatagrams()) {
...
 ```
 
 ## Detailed design discussion
 
-Any WebTransport can provide any of the following capabilities (mixins):
+WebTransport supports multiple protocols, each of which provide some of the
+following capabilities.
 
-- Unidirectional streams are indefintely long streams of bytes in one direction 
+- Unidirectional streams are indefintely long streams of bytes in one direction
```
Comment (ricea) [inline explainer.md:185]: s/indefintely /indefinitely/

File: explainer.md [hunk 6]
```diff
@@ -163,41 +168,46 @@ for await (const datagram of transport.receiveDatagrams()) {
...
   with back pressure applied
   to the sender when either the receiver can't read quickly enough or when
   constrained by network capacity/congestions.  Useful for sending messages that
   do not expect a response.  In-order, reliable messaging can be achieved by
   sending many messages in a single stream. Out-of-order messaging can be achieved
   by sending one message per stream.
 
-- Bidirectional streams are like unidirectional streams, but in two directions.  
+- Bidirectional streams are like unidirectional streams, but in two directions.
```
Comment (ricea) [inline explainer.md:193]: Maybe we should explicitly say that they are full duplex?

File: explainer.md [hunk 7]
```diff
@@ -163,41 +168,46 @@ for await (const datagram of transport.receiveDatagrams()) {
...
-- Datagrams are small, out-of-order, unreliable messages.  They are useful for 
+- Datagrams are small, out-of-order, unreliable messages.  They are useful for
   sending messages with less API complexity
   and less network overhead than streams.
 
-A QuicTransport is a WebTransport that maps directly to QUIC streams and
-datagrams, which makes it easy to connect to servers that speak QUIC with
-minimum overhead.  It supports all of these capabilities.
+[QuicTransport](https://tools.ietf.org/html/draft-vvv-webtransport-quic)
+is a WebTransport that maps directly to QUIC streams and datagrams, which makes
```
Comment (ricea) [inline explainer.md:201]: I found "a WebTransport" to be confusing here. Can we say "a WebTransport protocol"?

File: explainer.md [hunk 8]
```diff
@@ -163,41 +168,46 @@ for await (const datagram of transport.receiveDatagrams()) {
...
+it easy to connect to servers that speak QUIC with minimum overhead.
```
Comment (ricea) [inline explainer.md:202]: s/minimum/minimal/

**Review by ricea — COMMENTED** [review]
Just some nits.

File: explainer.md (hunk 1 above)
Comment (yutakahirano) [reply explainer.md]: Done.

File: explainer.md (hunk 3 above)
Comment (yutakahirano) [reply explainer.md]: I don't know. It looks like a mistake, so I removed it.

File: explainer.md (hunk 7 above)
Comment (yutakahirano) [reply explainer.md]: Done.

File: explainer.md (hunk 8 above)
Comment (yutakahirano) [reply explainer.md]: Done.

File: explainer.md (hunk 6 above)
Comment (yutakahirano) [reply explainer.md:192]: Something like this?

> Bidirectional streams are full-duplex streams. A bidirectional stream is effectively a pair of unidirectional streams.

File: explainer.md (hunk 2 above)
Comment (yutakahirano) [reply explainer.md:157]: Yeah, we can use https://url.spec.whatwg.org/#concept-url-parser.

**Comment by yutakahirano** [comment]
Addressed comments. Do you have further comments?

File: explainer.md [hunk 9]
```diff
@@ -131,9 +132,13 @@ function deserializeNotification(serializedNotification) { ... }
...
-const transport = new QuicTransport('example.com', 10001);
-for await (const stream of transport.receiveBidirectionalStreams()) {
-  const notification = new Notification(deserializeNofitication(await stream.arrayBuffer()));
+const transport = new WebTransport('quic-transport://example.com:10001/path');
+for await (const stream of transport.incomingBidirectionalStreams) {
+  const buffers = []
+  for await (const buffer of stream.readable) {
+    buffers.push(buffer)
+  }
+  const notification = new Notification(deserializeNofitication(buffers));
```
Comment (ricea) [inline explainer.md:141]: `deserializeNotification`

File: explainer.md (hunk 6 above)
Comment (ricea) [reply explainer.md:192]: Yes.

**Review by ricea — APPROVED** [review]
lgtm with nits

File: explainer.md [hunk 10]
```diff
@@ -58,22 +58,22 @@ encrypted and congestion-controlled communication.
...
-1. A set of generic transport mixins that can be provided by any transport,
+1. A generic transport interface that can be provided by any transport,
    but match closely with QUIC's capabilities.
 
-2. A specific transport based on QUIC that implements all of the transport mixins.
+2. The transport interface can talk [a QUIC based protocol](https://tools.ietf.org/html/draft-vvv-webtransport-quic).
 
-3. A specific transport based on HTTP/3 that allows a subset of the transport
-   mixins able to be pooled with HTTP traffic (sharing a congestion control context).
+3. The transport interface can talk [a HTTP/3 based protocol](https://tools.ietf.org/html/draft-vvv-webtransport-http3)
```
[the remark this answers is not in the document]
Comment (yutakahirano) [reply explainer.md]: Done.

File: explainer.md [hunk 11]
```diff
@@ -58,22 +58,22 @@ encrypted and congestion-controlled communication.
...
 ## Example of sending unreliable game state to server using QUIC datagrams
 
 ```javascript
 // The app provides a way to get a serialized state to send to the server
 function getSerializedGameState() { ... }
 
-const transport = new QuicTransport('example.com', 10001);
-const datagramWriter = transport.sendDatagrams().getWriter();
+const transport = new WebTransport('quic-transport://example.com:10001/path');
+const datagramWriter = transport.datagramsWritable.getWriter();
```
[the remark this answers is not in the document]
Comment (yutakahirano) [reply explainer.md]: Done.

File: explainer.md [hunk 12]
```diff
@@ -86,10 +86,10 @@ setInterval(() => {
 // The app provides a way to get a serialized state to send to the server.
 function getSerializedGameState() { ... }
 
-const transport = new QuicTransport('example.com', 10001);
+const transport = new WebTransport('quic-transport://example.com:10001/path');
 setInterval(async () => {
   const message = getSerializedGameState();
-  const stream = await transport.createSendStream();
+  const stream = await transport.createUnidirectionaltream();
```
[the remark this answers is not in the document]
Comment (yutakahirano) [reply explainer.md]: Done.

File: explainer.md (hunk 9 above)
Comment (yutakahirano) [reply explainer.md]: Done.

**Review by vasilvv — APPROVED** [review]

**Pull request merged**